galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/routes/AuthTokensRoute.java

```diff
@@ -76,36 +76,76 @@ public AuthTokensRoute(
+        String loginId = queryParams.getSingleString(AuthenticationServlet.QUERY_PARAM_LOGIN_ID, null);
+        
+        if(queryParams.checkAtLeastOneQueryParameterPresent("loginId")){
+            validateLoginId(loginId, pathInfo);        
+        }
+
+        if(loginId == null){
+            tokensToReturn = handleGetAllTokens();
+        }
+        else{
+            tokensToReturn = handleGetTokensByLoginId(loginId);
+        }
+
+        return getResponseBuilder().buildResponse(request, response, "application/json",
+                getTokensAsJsonString(tokensToReturn), HttpServletResponse.SC_OK);
+    }
+
+    public List<AuthToken> handleGetAllTokens() throws ServletException, IOException, FrameworkException{
 
         List<AuthToken> tokensToReturn = new ArrayList<>();
         try {
-            // Retrieve all the tokens and put them into a mutable list before sorting them based on their creation time
+            // Retrieve all the tokens and put them into a mutable list before sorting them
+            // based on their creation time
             List<IInternalAuthToken> tokens = new ArrayList<>(authStoreService.getTokens());
             Collections.sort(tokens, Comparator.comparing(IInternalAuthToken::getCreationTime));
 
-            // Convert the token received from the auth store into the token bean that will be returned as JSON
-            for (IInternalAuthToken token : tokens) {
-                User user = new User(token.getOwner().getLoginId());
-                tokensToReturn.add(new AuthToken(
-                    token.getTokenId(),
-                    token.getDescription(),
-                    token.getCreationTime(),
-                    user)
-                );
-            }
+            // Convert the token received from the auth store into the token bean that will
+            // be returned as JSON
+            tokensToReturn = convertAuthStoreTokenIntoTokenBean(tokens);
+
         } catch (AuthStoreException e) {
             ServletError error = new ServletError(GAL5053_FAILED_TO_RETRIEVE_TOKENS);
             throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e);
         }
 
-        return getResponseBuilder().buildResponse(request, response, "application/json", getTokensAsJsonString(tokensToReturn), HttpServletResponse.SC_OK);
+        return tokensToReturn;
+
+    }
+
+    public List<AuthToken> handleGetTokensByLoginId(String loginId)
+            throws ServletException, IOException, FrameworkException {
+
+        logger.info("fetching access tokens by loginId");
+
+        List<AuthToken> tokensToReturn = new ArrayList<>();
+
+        try{
+            List<IInternalAuthToken> tokens = new ArrayList<>(authStoreService.getTokensByLoginId(loginId));
+
+        // Convert the token received from the auth store into the token bean that will
+        // be returned as JSON
+        tokensToReturn = convertAuthStoreTokenIntoTokenBean(tokens);
```

Comment:
This is duplicated in both the `getAllTokens` and `getTokensByLoginId` methods - would be good to pull it out into the `handleGetRequest` method to be called before writing the response